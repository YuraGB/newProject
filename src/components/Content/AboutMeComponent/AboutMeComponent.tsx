/**
 * AboutMeComponent
 *
 * @author Yurii Huriianov <yuhur1985@gmail.com
 * @copyright 2020
 */

import React from 'react';
import ContentSection from "../ContentSection/ContentSection";

/**
 * AboutMeComponent
 * @return {*} JSX.Element
 */
const AboutMeComponent:React.FC = (): JSX.Element => {
    return (
        <ContentSection title="About Me">
            <p>
                Front-end web developer with over 5 years of experience in e-commerce projects.
                I have deep knowledge's in ReactJs, JS, HTML, Node js.
                Experienced in creating React components, Magento 2 modules,
                and creating highly effective user-friendly web interfaces.
                My greatest strength is responsibility, teamwork, hard-worker.
            </p>
        </ContentSection>
    )
};

export default AboutMeComponent;