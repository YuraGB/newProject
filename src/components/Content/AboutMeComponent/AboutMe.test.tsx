import React from "react";
import { screen, render } from "@testing-library/react";
import AboutMeComponent from "./AboutMeComponent";
import renderer from "react-test-renderer";

describe("AboutMeComponent tests", () => {
  const data = "test string for about me component";
  const SECTION_TITLE = "About Me";

  test("About me component with no props data", async () => {
    render(<AboutMeComponent />);
    const title = screen.queryByText(SECTION_TITLE);
    expect(title).not.toBeInTheDocument();
  });

  test("App loads resume data", async () => {
    render(<AboutMeComponent about_me={data} />);

    const aboutMeTitle = screen.queryByText(SECTION_TITLE);

    expect(aboutMeTitle).toBeInTheDocument();
  });

  test("match snapshots about me component", () => {
    // @ts-ignore
    const tree = renderer.create(<AboutMeComponent about_me={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
