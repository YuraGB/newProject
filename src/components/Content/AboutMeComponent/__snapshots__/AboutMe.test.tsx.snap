// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMeComponent tests match snapshots about me component 1`] = `
<section
  className="Section"
>
  <h5>
    About Me
  </h5>
  <hr />
  <p>
    test string for about me component
  </p>
</section>
`;
